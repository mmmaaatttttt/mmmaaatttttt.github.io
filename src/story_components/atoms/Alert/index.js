import PropTypes from "prop-types";
import styled, { css } from "styled-components";
import COLORS from "utils/styles";

const Alert = styled.div`
  background-color: ${props => props.color};
  padding: 0.5rem;
  border-radius: 0.25rem;
  width: 100%;

  ${props => props.center && css`
    text-align: center;
  `}
`

Alert.propTypes = {
  center: PropTypes.bool.isRequired,
  color: PropTypes.string.isRequired
};

Alert.defaultProps = {
  center: false,
  color: COLORS.GREEN
};

export default Alert;