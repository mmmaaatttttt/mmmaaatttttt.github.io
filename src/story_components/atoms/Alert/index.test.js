import React from "react";
import { render } from "@testing-library/react";
import Alert from "./";

it("renders without crashing", () => {
  render(<Alert />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<Alert />);
  expect(asFragment()).toMatchSnapshot();
});
