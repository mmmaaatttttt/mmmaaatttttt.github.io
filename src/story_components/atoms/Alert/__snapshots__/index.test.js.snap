// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="sc-bdVaJa DtitT"
    color="#52A081"
  />
</DocumentFragment>
`;
